### ISSUE TITLE
Can't name function argument $url on wire:click

### ISSUE THREAD

**Issue by froberts0n** [livewire/livewire#697]
Hi!
I've had the strangest bug, trying yo follow the example in the following artice : [Search with Laravel Livewire](https://medium.com/@branick/search-with-laravel-livewire-cb6dcd4ad541) by Barnick Weix to try out Livewire. The pagination links use the `wire:click` attribute to call a function in the component and I wanted to get the link URL back so I used `wire:click="setPage('{{ $url }}')"`. Here's the code of the function:
```php
public function setPage($url)
    {
        $this->currentPage = explode('page=', $url)[1];

        Paginator::currentPageResolver(function () {
            return $this->currentPage;
        });
    }
```
But every time I tried, I ended up with an Exception error saying that `explode() expects parameter 2 to be string, object given`. I checked the HTML in the browser and everything was fine : `<a href="#" class="page-link" wire:click="setPage('http://livewire.test/search?page=2')">2</a>`. However, when I tried to check the parameter in the code using `dd()`, the `$url` was always an instance of `Illuminate\Routing\UrlGenerator` instead of a string.
I solved the issue by renaming the parameter `$toto` instead of `$url` and everything worked!

I don't think this is a normal behavior...

**Comment by ninthspace** [comment]
The article was written before Livewire had its own pagination support, so things may have changed since. Have a look here: https://laravel-livewire.com/docs/pagination

**Comment by froberts0n** [comment]
> The article was written before Livewire had its own pagination support, so things may have changed since. Have a look here: https://laravel-livewire.com/docs/pagination

Thanks for the reply. However, I don't see how this relates to pagination. The problem is simply regarding the behavior of the argument when its name is changed.

Here, naming the variable `$url` makes Livewire interprets the string passed by the element and replaces it with a ` UrlGenerator` object just because of the argument's name.

**Comment by calebporzio** [comment]
Yeah, this is a legitimate bug.

The problem lies in this method: `src/ComponentConcerns/HandlesActions.php@resolveActionParameters`

I tried fixing this by using Laravel's `app()->call()` for the dependency injection, but I actually found a (different) bug with that too: (default params are broken)
<img width="798" alt="Screen Shot 2020-03-20 at 12 18 13 PM" src="https://user-images.githubusercontent.com/3670578/77186989-2d95d480-6aaa-11ea-9b5d-429ef60f95e3.png">

I can't really sink too much more time into this right now, but it needs to be fixed. Does anyone have the capacity to work on it?
